### ISSUE TITLE
TypeError: promise.finally is not a function

### ISSUE THREAD

**Issue by dhanir** [danialfarid/ng-file-upload#1575]
I'm getting this error:

> TypeError: promise.finally is not a function

when uploading images. The upload works fine, but the code never returns to my .then().

My code is as follows:
```
Upload.upload({
	url: "/umbraco/surface/newspaper/createcompetitionnewspaper",
	file: image,
	data: {
		name: newspaper.Name,
		description: newspaper.Description,
		sponsor: newspaper.Sponsor,
		brand: newspaper.Brand,
		openingDate: newspaper.OpeningDate,
		startDate: newspaper.StartDate,
		endDate: newspaper.EndDate,
		numberOfPages: pages,
		frontpageTemplates: frontTemplates,
		backPageTemplates: backTemplates
	}
}).then(function (response) {
	return "Success: " + response.config.data.file.name + " uploaded";
}, function (response) {
	console.log(response);
	var exception = {
		title: "Error: " + response.config.data.file.name + " failed",
		message: response.data.ExceptionMessage,
		type: response.data.ExceptionType,
		stackTrace: response.data.StackTrace,
	};

	return exception;
});
```
Anyone got a suggestion on how to resolve this?

**Comment by danialfarid** [comment]
This doesn't seem to be related to this plugin. Find which line the error "TypeError" is and it is probably from another plugin.

**Comment by vmburd** [comment]
I'm getting the same error.

Here is my code

```
$scope.uploadFile = function () {
	Upload.upload({
		url : 'backoffice/AudioFile/AudioFileApi/UploadFileToServer',
		file : $scope.data.uploadThisFile,
		data : {
			path : $scope.data.defaultPath
		}
	}).then(function (resp) {
		if (resp.config.data.notifications && resp.config.data.notifications.length > 0) {
			console.log("File import failed. " + data.notifications[0].message);
		} else {
			console.log('Success ' + resp.config.data.file.name + 'uploaded. Response: ' + resp.data);
		}
	}, function (resp) {
		console.log("File import failed. " + resp.status);
	}, function (evt) {
		// calculate progress in percentage
		var progressPercentage = parseInt(100.0 * evt.loaded / evt.total, 10);

		// set percentage property on file
		$scope.data.uploadProgress = progressPercentage;

		// set uploading status on file
		$scope.data.uploadStatus = "uploading";
	});
}
```

And here is the error message  - as you can see the sendHttp method in ng-file-upload seems to be the culprit  with this code

```
    promise['finally'](function () {
      upload.promisesCount--;
    });
```

> TypeError: promise.finally is not a function
>     at sendHttp (ng-file-upload.js?cdv=117268614:188)
>     at upload (ng-file-upload.js?cdv=117268614:330)
>     at Object.$scope.uploadFile (audioFilePicker.controller.js?cdv=117268614:107)
>     at angular.min.js?cdv=117268614:74
>     at angular-mobile.js?cdv=117268614:270
>     at Object.$eval (angular.min.js?cdv=117268614:92)
>     at Object.$apply (angular.min.js?cdv=117268614:92)
>     at HTMLButtonElement.<anonymous> (angular-mobile.js?cdv=117268614:269)
>     at HTMLButtonElement.dispatch (jquery.min.js?cdv=117268614:5)
>     at HTMLButtonElement.y.handle (jquery.min.js?cdv=117268614:5)(anonymous function) @ angular.min.js?cdv=117268614:63(anonymous function) @ angular.min.js?cdv=117268614:54$apply @ angular.min.js?cdv=117268614:92(anonymous function) @ angular-mobile.js?cdv=117268614:269dispatch @ jquery.min.js?cdv=117268614:5y.handle @ jquery.min.js?cdv=117268614:5

**Comment by pamuditha** [comment]
i got same error 
angular.js:6173 TypeError: promise.finally is not a function
    at sendHttp (http://localhost:9000/bower_components/ng-file-upload/ng-file-upload.js:188:23)
    at upload (http://localhost:9000/bower_components/ng-file-upload/ng-file-upload.js:330:12)
    at http://localhost:9000/js/controllers.js:524:32
    at Array.forEach (native)
    at Object.forEach (https://ajax.googleapis.com/ajax/libs/angularjs/1.1.4/angular.js:151:11)
    at Object.$scope.uploadFiles (http://localhost:9000/js/controllers.js:522:15)
    at https://ajax.googleapis.com/ajax/libs/angularjs/1.1.4/angular.js:6801:19
    at update (http://localhost:9000/bower_components/ng-file-upload/ng-file-upload.js:538:27)
    at http://localhost:9000/bower_components/ng-file-upload/ng-file-upload.js:608:15
    at https://ajax.googleapis.com/ajax/libs/angularjs/1.1.4/angular.js:10320:28

**Comment by hellofreewill** [comment]
i got the same error:
angular.min.js:107 TypeError: Upload.upload is not a function
    at m.$scope.uploadFile (uploadController.js:41)
    at fn (eval at compile (angular.min.js:210), <anonymous>:4:323)
    at f (angular.min.js:250)
    at m.$eval (angular.min.js:134)
    at m.$apply (angular.min.js:134)
    at HTMLButtonElement.<anonymous> (angular.min.js:250)
    at HTMLButtonElement.dispatch (jquery.min.js:3)
    at HTMLButtonElement.r.handle (jquery.min.js:3)

my code:
 $scope.uploadFile = function(file) {
                console.log("uploading...");
                Upload.upload({
                    url: 'https://angular-file-upload-cors-srv.appspot.com/upload',
                    method: 'POST',
                    data: {file: $scope.file},
                }).then(function (response) {
                    $timeout(function () {
                        file.result = response.data;
                    });
                }, function (response) {
                    if (response.status > 0)
                        $scope.errorMsg = response.status + ': ' + response.data;
                }, function (evt) {
                    // Math.min is to fix IE which reports 200% sometimes
                    file.progress = Math.min(100, parseInt(100.0 * evt.loaded / evt.total));
                });
            }
<form name="myForm">
    <fieldset>
				<legend>Upload file in category level (.xlsx) : </legend>
				<input type="file" ngf-select ng-model="xlsFile" name="file"
					   accept="application/vnd.ms-excel" ngf-max-size="100MB" required
					   ngf-model-invalid="errorFile">
				<i ng-show="myForm.file.$error.required">*required</i><br>
				<i ng-show="myForm.file.$error.maxSize">File too large
					{{errorFile.size / 1000000|number:1}}MB: max 100M</i>
				<button ng-disabled="!myForm.$valid"
						ng-click="uploadFile(xlsFile)">Submit</button>
				<span ng-show="xlsFile.progress >= 0">
        			<div ng-bind="xlsFile.progress + '%'"></div>
      			</span>
				<span ng-show="xlsFile.result">Upload Successful</span>
				<span class="err" ng-show="errorMsg">{{errorMsg}}</span>
			</fieldset>
    <br>
  </form>

**Comment by danialfarid** [comment]
Create a jsfiddle for the issue. I need the angular and plugin version and steps to reproduce.

**Comment by danialfarid** [comment]
Fixed at 12.2.0
